.github/workflows/rust.yml

```
@@ -16,15 +16,31 @@ jobs:
       checks: write
 
     steps:
-      - uses: actions/checkout@v1
+      - uses: actions/checkout@v2
         with:
             submodules: recursive
+      - name: Rust Toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+            profile: minimal
+            toolchain: stable
+            components: rustfmt, clippy
+            override: true
       - name: Build
-        run: cargo build --verbose
-      - name: Run tests
-        run: cargo test --verbose
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: --verbose
```

formatting nitpick for consistency with the surrounding content
```suggestion
            command: build
            args: --verbose
```